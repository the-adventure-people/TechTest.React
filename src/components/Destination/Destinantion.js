import styles from "./Destination.module.scss";

export default function Destination({ destination }) {
  const { name, image, country, continent } = destination;

  return (
    <div className={styles.tile}>
      <img
        src={`${image}?auto=compress&auto=format&fit=crop&h=400&max-w=400`}
        alt={country}
      />
      <h2>{name}</h2>
      <p>{country}</p>
      <p>{continent}</p>
    </div>
  );
}
