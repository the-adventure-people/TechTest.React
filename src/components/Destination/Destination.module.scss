.tile {
    
    & img {
        max-width: 100%;
    }
}