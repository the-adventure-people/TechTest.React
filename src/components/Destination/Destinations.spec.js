import React from "react";
import { render } from "@testing-library/react";
import Destination from "./Destinantion";

describe("Destination component", () => {
    it("Should render the card", () => {
        const mockData ={
            name: "Angkor Wat",
            image:
              "https://imgix.cosmicjs.com/06e2bcd0-d2e4-11e9-9ea8-2b9d842385af-the-adventure-people-destination-cambodia.jpg",
            country: "Cambodia",
            continent: "Asia",
        };
        const {baseElement} = render(<Destination destination={mockData} />);
        expect(baseElement).toMatchSnapshot();
    })
})