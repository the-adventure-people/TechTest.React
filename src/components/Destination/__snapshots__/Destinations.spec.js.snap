// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Destination component Should render the card 1`] = `
<body>
  <div>
    <div
      class="tile"
    >
      <img
        alt="Cambodia"
        src="https://imgix.cosmicjs.com/06e2bcd0-d2e4-11e9-9ea8-2b9d842385af-the-adventure-people-destination-cambodia.jpg?auto=compress&auto=format&fit=crop&h=400&max-w=400"
      />
      <h2>
        Angkor Wat
      </h2>
      <p>
        Cambodia
      </p>
      <p>
        Asia
      </p>
    </div>
  </div>
</body>
`;
